models/user_context.go

```diff
@@ -0,0 +1,10 @@
+package models
+
+import "github.com/google/uuid"
+
+type UserContext struct {
```

Comment:
Ensure that the `UserContext` struct does not expose sensitive information unintentionally and that it adheres to the principle of least privilege. Only include necessary fields required for downstream processing. [important]